fix: initial efforts to create useful, standard readme with purpose. (#30863)
openedx/edx-platform: README.rst

############

> edX Platform

Reviewer: I know this is the name of the repo, but the way it's written here I feel undercuts our attempts to consistently say "Open edX Platform".

I'd suggest something like, "Open edX Platform", "The Open edX Platform", or "edx-platform". You could also add the word "monolith" in there, or other descriptors. I don't think what we say in the readme title necessarily needs to exact match the name of the repo.

############
| |License: AGPL v3| |Status| |Python CI|

.. |License: AGPL v3| image:: https://img.shields.io/badge/License-AGPL_v3-blue.svg
  :target: https://www.gnu.org/licenses/agpl-3.0

.. |Python CI| image:: https://github.com/openedx/edx-platform/actions/workflows/ci.yml/badge.svg
  :target: https://github.com/openedx/edx-platform/actions/workflows/ci.yml

.. |Status| image:: https://img.shields.io/badge/status-maintained-31c653

> Purpose

Reviewer 2: This sounds like a good summary to me <3

> =======

Reviewer 3: ```suggestion
-------
```
As is, this is creating a new heading level and nesting the rest of the README under `Purpose`.

> The Open edX platform is a service oriented platform for authoring and

Reviewer 3: ```suggestion
The `Open edX Platform <https://openedx.org>`_ is a service-oriented platform for authoring and
```
I think this is a good place to link to openedx.org. Also a couple grammar nits.

delivering online learning at any scale.  The platform is written in
Python and JavaScript and makes extensive use of the Django
framework. At the highest level, the platform is composed of a
monolith, some independently deployable applications (IDAs), and
micro-frontends (MFEs) based on the ReactJS.

This repository hosts the monolith at the center of the Open edX

> platform.  Functionally, the edx-platform repository provides two services:

Reviewer 3: Whoops, sorry @e0d , one more thing -- you need a newline here for the RST bullet points to render.

```suggestion
platform.  Functionally, the edx-platform repository provides two services:

```

> platform.  Functionally, the edx-platform repository provides the
> environment for authoring courses, Open edX Studio, also called the
> CMS, and the course delivery service, called the LMS.

Reviewer 3: ```suggestion
platform.  Functionally, the edx-platform repository provides two services:
* CMS, which powers Open edX Studio, the platform's learning content authoring environment; and
* LMS, which delivers learning content.
```
just some pedantry around courses vs other learning content as well as [CMS vs Studio](https://github.com/openedx/edx-platform/blob/master/docs/decisions/0013-cms-vs-studio.rst).

Reviewer: Could we define the acronyms CMS and LMS? I think it sticks better when people know what it means. They are not so commonly used to be universally known, I think.

> * CMS, which powers Open edX Studio, the platform's learning content authoring environment; and
> * LMS, which delivers learning content.

Reviewer 3: Suggestion to address @sarina 's point about acronyms:

```suggestion
* CMS (Content Management Service), which powers Open edX Studio, the platform's learning content authoring environment; and
* LMS (Learning Management Service), which delivers learning content.
```
